### ISSUE TITLE
[MAIN] [Pipeline] Collate loss functions

### ISSUE THREAD

**Issue by Mars-Martiny** [Mars-Martiny/CITS5206_group1#25]
Gather loss functions to use in training loop.
Adjust to integrate with current task and account for modularity requirements.
